File: File_Operations/FileRead.java
import java.io.*;

public class FileRead {

	public static String readFile(String input) {

		try {
			FileReader fr = new FileReader(input);
			BufferedReader in = new BufferedReader(fr);
			String output = "";
			String line = null;
			while((line = in.readLine()) != null) {
				output = output + line + "\n";
			}
			return output;
		}
		catch(IOException e) {
			System.out.println("Error occurred while trying to read " + input);
		}
		return "";
	}

	public static void main(String[] args) {
		String out = readFile("read.txt");
		System.out.println(out);
	}

}

/*
	Running from command line:
		1) javac FileRead.java <--- compile java file to class file
		2) java FileRead <--- Execute compiled Java file
*/

File: File_Operations/FileRead.cpp
#include <iostream>
#include <fstream>

using namespace std;

string readFile(string file) {
	string line;
	string output;
	ifstream fileReader (file);
	if(fileReader.is_open()){
		while(getline(fileReader, line)){
			output = output + line + "\n";
		}
		fileReader.close();
	}
	return output;
}

int main(){
	string file = "read.txt";
	string output = readFile(file);
	cout << output;
	return 1;
}

/*
	Running from command line:
		1) Make <--- compile Makefile
		2) ./runFileRead.cpp <--- Execute compiled C++ file
*/

